app/models/health/qualifying_activity.rb

```diff
@@ -510,21 +570,50 @@ def any_submitted_of_type_for_day_for_patient?
       same_of_type_for_day_for_patient.submitted.exists?
     end
 
-    def occurred_after_enrollment?
-      date_of_activity.present? && patient&.patient_referral.present? && date_of_activity >= patient.patient_referral.enrollment_start_date
+    def occurred_during_any_enrollment?
+      date_of_activity.present? && patient.patient_referrals.active_within_range(start_date: date_of_activity, end_date: date_of_activity).exists?
     end
 
-    def occurred_within_three_months_of_enrollment
-      date_of_activity.present? && patient&.patient_referral.present? && patient&.enrollment_start_date.present? && date_of_activity <= (patient.outreach_cutoff_date)
+    def occurred_during_enrollment?
+      date_of_activity.present? && patient.patient_referrals.contributing.active_within_range(start_date: date_of_activity, end_date: date_of_activity).exists?
+    end
+
+    def occurred_within_three_months_of_enrollment?
+      date_of_activity.present? && patient.current_days_enrolled <= 90
```

Comment:
I think this is probably OK, but I know there were some places where the actual months mattered.  Are we sure 90 days is a safe proxy?

Reply:
As I read the revised rules, time spent dis-enrolled doesn't count towards the clock... So, I think so...